### DISCUSSION TITLE
Error within the example code_browser.py

### DISCUSSION THREAD

**Discussion by klerai** [Textualize/textual#2664] [Q&A] [answered]
The example code returns with error, when selecting a file to display:
    ../python3.8/site-packages/rich/text.py:927 in append                                        │
    TypeError: Only str or Text can be appended to Text

caused by line 69
   self.sub_title = str(event.path)

The following solution works for me:
    self.sub_title = event.path.name

**Comment by TomJGooding** [comment]
How exactly are you running the `code_browser.py` example? What version of Textual do you have installed?

I'm not seeing this error after running the example within my clone of Textual.

**Comment by klerai** [comment]
- I am calling the app from the CLI:
python code_browser.py

- here is an output snip-set from the CLI:
textual diagnose

# Textual Diagnostics

## Versions

| Name    | Value  |
|---------|--------|
| Textual | 0.24.1 |
| Rich    | 13.3.5 |

## Python

| Name           | Value                                                          |
|----------------|----------------------------------------------------------------|
| Version        | 3.8.16                                                         |
| Implementation | CPython                                                        |
| Compiler       | GCC 11.3.0                                                     |
| Executable     | /home/klerai/.config/jupyterlab-desktop/jlab_server/bin/python |

## Operating System

| Name    | Value                                                        |
|---------|--------------------------------------------------------------|
| System  | Linux                                                        |
| Release | 5.14.21-150400.24.63-default                                 |
| Version | #1 SMP PREEMPT_DYNAMIC Tue May 2 15:49:04 UTC 2023 (fd0cc4f) |

## Terminal

**Reply by davep** [reply]
As with Tom, I'm not seeing this issue too. I note that you're using a slightly older version of Textual too. First off I'd suggest trying this again with the latest release of Textual (0.26.0 as of the time of writing). If you still get the issue could you post the full stacktrace please?

**Comment by klerai** [comment, marked as answer]
I've checked your most current branch. Your code has not changed. As I've mentioned in my original message, a change from event.path (which returns a path object) to event.path.name (returning a correct string object) was my working solution. Thanks for your efforts, mates.

**Reply by davep** [reply]
There have been [changes to the handling of different types when it comes to setting the application title](https://github.com/Textualize/textual/pull/2532), hence my suggestion that it would be worth testing with the latest release of Textual. At a quick glance it's not obvious what the cause of the problem is, but given neither Tom nor I can recreate it with the current release of Textual it's a good idea to ensure we're all looking at the same thing.

**Reply by willmcgugan** [reply]
Version [0.25.0](https://github.com/Textualize/textual/blob/main/CHANGELOG.md#0250---2023-05-17) introduced the ability to set title and sub title to anything, by stringifying. @klerai we update regularly, so its easy to get out of sync.

**Comment by klerai** [comment]
Dear Dave,

thanks for your suggestion. I just recently came across your library and installed a version via PIP. So I am just earlier student from your work.

But growing up with a terminal in the late seventies, I am very impressed with what your team has achieved.

Warmly,

Rainer
